src/client/eppo-client-with-bandits.spec.ts

```diff
@@ -40,16 +49,212 @@ describe('EppoClient Bandits E2E test', () => {
     const httpClient = new FetchHttpClient(apiEndpoints, 1000);
     const configurationRequestor = new ConfigurationRequestor(httpClient, flagStore, banditStore);
     await configurationRequestor.fetchAndStoreConfigurations();
+  });
 
+  beforeEach(() => {
     client = new EppoClient(flagStore, banditStore);
     client.setIsGracefulFailureMode(false);
+    client.setAssignmentLogger({ logAssignment: mockLogAssignment });
+    client.setBanditLogger({ logBanditAction: mockLogBanditAction });
+    jest.clearAllMocks();
   });
 
   afterAll(() => {
     jest.restoreAllMocks();
   });
 
-  it('TODO', () => {
-    // TODO: bandit test cases with the client
+  describe('Shared test cases', () => {
+    const testData = readTestData<BanditTestCase>(BANDIT_TEST_DATA_DIR);
+    // Build a map for more useful test names
+    const testsByFlagKey: Record<string, BanditTestCase> = {};
+    testData.forEach((testCase) => (testsByFlagKey[testCase.flag] = testCase));
+
+    it.each(Object.keys(testsByFlagKey))(
+      'Shared bandit test case - %s',
+      async (flagKey: string) => {
+        const { defaultValue, subjects } = testsByFlagKey[flagKey];
+        let numAssignmentsChecked = 0;
+        subjects.forEach((subject) => {
+          // TODO: handle already-bucketed attributes
+          // TODO: common test case with a numeric value passed as a categorical attribute and vice verse
+
+          const actions: Record<string, Attributes> = {};
+          subject.actions.forEach((action) => {
+            actions[action.actionKey] = {
+              ...action.numericAttributes,
+              ...action.categoricalAttributes,
+            };
+          });
+
+          const subjectAttributes = {
+            ...subject.subjectAttributes.numeric_attributes,
+            ...subject.subjectAttributes.categorical_attributes,
```

Comment:
Is this the typo in the shared test file showing up? I prefer if we fix it there ASAP before cementing the typo everywhere. (EDIT: I saw that you have a comment about this later on.)

Reply:
It is! A plan to fix it and add some tests after merging this (into parent branch, not main)